File: 01/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

func getList(inputFile string) []int {
	fInput, _ := os.Open(inputFile)
	defer fInput.Close()

	currentCalories := 0
	var elfList []int
	scanner := bufio.NewScanner(fInput)
	for scanner.Scan() {
		line := scanner.Text()
		if line == "" {
			elfList = append(elfList, currentCalories)
			currentCalories = 0
		} else {
			iCalories, _ := strconv.Atoi(line)
			currentCalories += iCalories
		}
	}

	sort.Ints(elfList)
	return elfList
}

func part1(elfList []int) int {
	return elfList[len(elfList)-1]
}

func part2(elfList []int) int {
	elfSlice := elfList[len(elfList)-3 : len(elfList)]

	sum := 0
	for _, calories := range elfSlice {
		sum += calories
	}
	return sum
}

func main() {
	elfList := getList("./input.txt")
	fmt.Printf("Part 1: %d\n", part1(elfList))
	fmt.Printf("Part 2: %d\n", part2(elfList))
}

File: 01/main.rs
use std::fs::File;
use std::io::{BufReader, BufRead, Error};

fn main() -> Result<(), Error> {
    let file = File::open("./input.txt")?;
    let reader = BufReader::new(file);

    let mut calories = Vec::new();
    let mut sum = 0;
    for line in reader.lines() {
        let num = line?.parse::<u32>().unwrap_or(0);
        if num == 0 {
            calories.push(sum);
            sum = 0;
        } else {
            sum += num;
        }
    }

    calories.sort();
    calories.reverse();

    println!("Part 1: {}", calories.first().unwrap());
    println!("Part 2: {}", calories.get(0..3).unwrap().iter().sum::<u32>());

    Ok(())
}